src/adapter/src/error.rs

```diff
@@ -221,9 +222,24 @@ pub enum AdapterError {
         found: Duration,
         name: String,
     },
-    /// An ALTER SOURCE referred to an upstream table that's already referred to.
-    SubsourceAlreadyReferredTo {
+    /// Trying to add a subsource whose name has been changed.
+    ///
+    /// At the time we introduce this error, we use these names to demultiplex
+    /// source outputs to multiple persist shards, so we can't handle this
+    /// ambiguity.
```

Comment:
This is really hard to understand...can you provide more context? What does it mean to 'introduce' this error?

Reply:
By "error", I meant "feature!" Will clarify this.

Reply:
Ah, I had meant "error," but I meant 'introduce' as in 'commit code that contains this enum member.' The point I'm trying to make is that we use `UnresolvedItemName` to resolve subsource output indices. If this ever drifts in the future, this is a bread crumb explaining why this is the way it is.